Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -449,10 +469,111 @@ class StoreKit1IntegrationTests: BaseBackendIntegrationTests {
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+    }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testPurchaseInDevicePostsReceipt() async throws {
+        try await self.purchaseProductFromStoreKit()
+
+        await asyncWait(
+            until: {
+                let entitlement = try? await Purchases.shared
+                    .customerInfo(fetchPolicy: .fetchCurrent)
+                    .entitlements[Self.entitlementIdentifier]
+
+                return entitlement?.isActive == true
+            },
+            timeout: .seconds(60),
+            pollInterval: .seconds(2),
+            description: "Entitlement didn't become active"
+        )
+    }
+
+}
+
+class StoreKit1ObserverModeIntegrationTests: BaseStoreKitObserverModeIntegrationTests {
+
+    func testPurchaseOutsideTheAppPostsReceipt() async throws {
+        try self.testSession.buyProduct(productIdentifier: Self.monthlyNoIntroProductID)
+
+        let info = try await Purchases.shared.restorePurchases()
+        try await self.verifyEntitlementWentThrough(info)
+    }
+
+}
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+class StoreKit2ObserverModeWithExistingPurchasesTests: StoreKit1ObserverModeWithExistingPurchasesTests {
+
+    override class var storeKit2Setting: StoreKit2Setting { return .enabledForCompatibleDevices }
+
+}
+
+/// Purchases a product before configuring `Purchases` to verify behavior upon initialization in observer mode.
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+class StoreKit1ObserverModeWithExistingPurchasesTests: BaseStoreKitObserverModeIntegrationTests {
+
+    // MARK: - Transactions observation
+
+    private static var transactionsObservation: Task<Void, Never>?
+
+    override class func setUp() {
+        Self.transactionsObservation?.cancel()
+        Self.transactionsObservation = Task {
+            // Silence warning in tests:
+            // "Making a purchase without listening for transaction updates risks missing successful purchases.
+            for await _ in Transaction.updates {}
+        }
+    }
+
+    override class func tearDown() {
+        Self.transactionsObservation?.cancel()
+        Self.transactionsObservation = nil
+    }
+
+    override func setUp() async throws {
+        // 1. Create `SKTestSession`
+        try self.configureTestSession()
+
+        // 2. Purchase product directly from StoreKit
+        try await self.purchaseProductFromStoreKit()
+
+        // 3. Configure SDK
+        try await super.setUp()
+    }
+
+    func testDoesNotSyncExistingPurchase() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let info = try await Purchases.shared.customerInfo(fetchPolicy: .fetchCurrent)
+        self.assertNoPurchases(info)
+    }
```

Comment:
this test case is interesting. ideally purchases don't go through before the SDK gets configured, otherwise, this is what happens - we miss the purchase. 

in real life the most common case is a renewal which happens with the SDK being closed, and in that case, it will still show up through the queue. Same for purchases on other devices.